ModelVale/ModelVale/ViewControllers/Main/ModelViewController.m

```diff
@@ -48,23 +50,30 @@ - (void)viewDidLoad {
     
     self.numClusters = 2;
     
-    //XXX todo configure healthbarview
-    [self.healthBarView initializeAnimationsWithDuration:animationDuration maxHealth:90 health:50];
+    [self.healthBarView initializeAnimationsWithDuration:animationDuration maxHealth:90 health:80];
     [self.healthBarView animateFillingHealthBar:self.healthBarView.healthPath layer:self.healthBarView.healthShapeLayer];
     
     [self setHealthBarPropsForXP];
     self.clusters = [self initializeXPClusters:self.numClusters avgNumPerCluster:20 seed:self.seed];
     [self animateXPClusters:self.clusters];
-    // call animations on those image views
+}
+
+- (void) viewWillAppear:(BOOL)animated {
+    [self reanimateXPClusters];
 }
 
 - (void) setHealthBarPropsForXP {
     self.seed = CGPointMake(self.view.frame.size.width - 15, self.view.frame.size.height -100);
     //XXX todo this calculation is slightly off (to the left and up too much)
-    CGPoint healthBarViewOrigin = [self.view convertPoint:self.healthBarView.bounds.origin fromView:self.healthBarView];
-    CGFloat filledHealthXCoord = self.healthBarView.leftTopPoint.x + self.healthBarView.barWidth * self.healthBarView.filledHealthWidthPercent;
-    CGFloat middleHealthYCoord = self.healthBarView.rightTopPoint.y + 0.5*self.healthBarView.barHeight;
-    self.XPEndPoint = CGPointMake(healthBarViewOrigin.x + filledHealthXCoord, healthBarViewOrigin.y + middleHealthYCoord);
+    // Hierarchy
+    // self.view --> self.detailsView --> self.avatarStackView --> self.healthBarView
+    CGPoint healthBarViewOrigin1 = [self.view convertRect:self.avatarStackView.frame fromView:self.detailsView].origin; // Is actually the avatarStackView origin
```

Comment:
Not sure if you are trying to get the origin of healthBarView (which is a subview of avatarStackView?) based on the coordinates of the VC view (self.view)? That's what I understand from your logic and comments here. If so I would do this: ie - 
    CGPoint healthBarViewOrigin1 = [self.avatarStackView convertRect:self.view.frame toView:self.view].origin; // Is actually the avatarStackView origin

See the second answer here: https://stackoverflow.com/questions/17572304/get-position-of-uiview-in-respect-to-its-superviews-superview